llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -1688,8 +1740,8 @@ class LoopVectorizationCostModel {
 
   /// Control finally chosen tail folding style. The first element is used if
   /// the IV update may overflow, the second element - if it does not.
-  std::pair<TailFoldingStyle, TailFoldingStyle> ChosenTailFoldingStyle =
-      std::make_pair(TailFoldingStyle::None, TailFoldingStyle::None);
+  std::pair<std::optional<TailFoldingStyle>, std::optional<TailFoldingStyle>>
```

Comment:
simpler to shift optional outside, if possible, i.e. both are either set or not set?